File: manejoArray.py
print("--Registro de personas--")
print("Si desea terminar escriba 'finalizar' en el campo nombre.")

personas = []
    

    
while True: 
        nombre = input("Nombre: ")
        if nombre.lower() == "finalizar":
            break
        edad = (input("Edad: "))
        nota = int(input("Nota: "))

        personas.append([nombre, edad, int(nota)])
      


print("Lista de personas: ")
for persona in personas:
    print(f"Nombre: {persona[0]}, Edad: {persona[1]}, Nota: {persona[2]}")


notas = sorted(personas, key=lambda x: int(x[2]), reverse=True)


print("\nLista por nota: ")
for persona in notas:
    print(f"Nombre: {persona[0]}, Edad: {persona[1]}, Nota: {persona[2]}")

File: manejoArray.js
console.log("---Registro de Personas---")

let personas = []

console.log("///Ingrese los siguientes datos///")
console.log("si desea terminar escriba 'finalizar' en el campo nombre")
while (true){
    let nombre = prompt("Nombre: ")
    if(nombre == "finalizar") {break;}
   
    let edad = prompt("Edad: ")
   
    let nota = prompt("Nota: ")
    personas.push({ nombre: nombre, edad: edad, nota: parseFloat(nota) });
};

console.log("Lista de Personas: ", personas);

personas.sort((a, b) => b.nota - a.nota);


console.log("Lista por Notas (de mayor a menor):");
console.log(personas);
